Added client_id for non embedded signature requests

neillom opened the pull request:
Added client_id for non embedded requests to enable white-labeling.

Fix for reported issue #40

neillom committed:
ADDING CLIENT_ID PARAM TO SIGNATURE REQUEST

Added client_id for non embedded requests to enable white-labeling.

Fix for reported issue #40

neillom committed:
UPDATED SEND SIGNATURE LOGIC

Logic to select API call was based on client_id not being null. This meant that all requests with client_id populated would create an embedded request.

Updated send_signature_request_embedded parameters to include identifier to determine the request type. Altered selection process in _send_signature_request to reflect this.

Fix for reported issue #40

martinytodorov commented on hellosign_sdk/hsclient.py:
```diff
@@ -1353,7 +1357,7 @@ def _check_required_fields(self, fields=None, either_fields=None):
 
     @api_resource(SignatureRequest)
-    def _send_signature_request(self, test_mode=False, client_id=None, files=None, file_urls=None, title=None, subject=None, message=None, signing_redirect_url=None, signers=None, cc_email_addresses=None, form_fields_per_document=None, use_text_tags=False, hide_text_tags=False, metadata=None, ux_version=None, allow_decline=False):
+    def _send_signature_request(self, test_mode=False, client_id=None, files=None, file_urls=None, title=None, subject=None, message=None, signing_redirect_url=None, signers=None, cc_email_addresses=None, form_fields_per_document=None, use_text_tags=False, hide_text_tags=False, metadata=None, ux_version=None, allow_decline=False, is_for_embedded_signing=None):
```
```suggestion
    def _send_signature_request(self, test_mode=False, client_id=None, files=None, file_urls=None, title=None, subject=None, message=None, signing_redirect_url=None, signers=None, cc_email_addresses=None, form_fields_per_document=None, use_text_tags=False, hide_text_tags=False, metadata=None, ux_version=None, allow_decline=False, is_for_embedded_signing=False):
```

martinytodorov reviewed (changes requested):
Hey Marc, I added one suggestion and because I previously merged a big PR yours is now in conflict.

Can you try to fix it, or may be just pull the new changes that were merged and add your changes again (might be faster).

Thanks for adding that!

neillom committed:
Merge branch 'v3' of https://github.com/hellosign/hellosign-python-sdk into v3

 Conflicts:
	hellosign_sdk/hsclient.py

neillom committed:
Fixes for issues reported in #40 and #58

Added client_id for non embedded signature requests to allow the use of
client_id for white-labelling

Updated logic to select correct API endpoint. The call was based on
client_id not being null. This meant that all requests with client_id
populated would create an embedded request. Updated
send_signature_request_embedded parameters to include identifier
to determine the request type. Altered selection process in
_send_signature_request to reflect this.

Included fix for issue #58. The latest merge meant that iuf the option
signing_options was empty this would cause a malformed JSON exception.
The json.dumps call was causing us to send null to the endpoint which
returned a 400 error. Updated the call to replace a null payload with
an empty JSON payload.

Fix for reported issue #40
Fix for reported issue #58

neillom commented:
Hey Martin, included fix for issue I spotted after yesterday's merge.

martinytodorov reviewed (approved):
Looks good. This `.replace('null','{}')` will fix the signing_options for now, but I am about to open another PR that will move this into one of those format functions in HSFormat.
